### ISSUE TITLE
webpage blank

### ISSUE THREAD

**Issue by bunnyyyyyy** [insidesherpa/JPMC-tech-task-2-PY3#55]
i am not getting any error when i did npm start and when the webpage opens it is just blank and i could find any error in that. i kinda out of idea on how to make it work now

![WhatsApp Image 2020-04-15 at 12 20 22 AM](https://user-images.githubusercontent.com/63678361/79276236-3f686d00-7eda-11ea-8a34-662131ce475f.jpeg)
![WhatsApp Image 2020-04-15 at 12 20 21 AM](https://user-images.githubusercontent.com/63678361/79276242-41323080-7eda-11ea-9fd5-ec7e0df77f93.jpeg)
i have attached the image here. thanks

**Comment by 9916103020** [comment]
@bunnyyyyyy could you show me your code please of Graph.tsx?

**Comment by bunnyyyyyy** [comment]
import React, { Component } from 'react';
import { Table } from '@jpmorganchase/perspective';
import { ServerRespond } from './DataStreamer';
import './Graph.css';

/**
 * Props declaration for <Graph />
 */
interface IProps {
  data: ServerRespond[],
}

/**
 * Perspective library adds load to HTMLElement prototype.
 * This interface acts as a wrapper for Typescript compiler.
 */
interface PerspectiveViewerElement {
  load: (table: Table) => void,
}

/**
 * React component that renders Perspective based on data
 * parsed from its parent through data property.
 */
class Graph extends Component<IProps, {}> {
  // Perspective table
  table: Table | undefined;

  render() {
    return React.createElement('perspective-viewer');
  }

  componentDidMount() {
    // Get element to attach the table from the DOM.
    const elem: PerspectiveViewerElement = document.getElementsByTagName('perspective-viewer')[0] as unknown as PerspectiveViewerElement;

    const schema = {
      stock: 'string',
      top_ask_price: 'float',
      top_bid_price: 'float',
      timestamp: 'date',
    };

    if (window.perspective && window.perspective.worker()) {
      this.table = window.perspective.worker().table(schema);
    }
    if (this.table) {
      // Load the `table` in the `<perspective-viewer>` DOM reference.

      // Add more Perspective configurations here.
      elem.load(this.table);
    }
  }

  componentDidUpdate() {
    // Everytime the data props is updated, insert the data into Perspective table
    if (this.table) {
      // As part of the task, you need to fix the way we update the data props to
      // avoid inserting duplicated entries into Perspective table again.
      this.table.update(this.props.data.map((el: any) => {
        // Format the data from ServerRespond to the schema
        return {
          stock: el.stock,
          top_ask_price: el.top_ask && el.top_ask.price || 0,
          top_bid_price: el.top_bid && el.top_bid.price || 0,
          timestamp: el.timestamp,
        };
      }));
    }
  }
}

export default Graph;

**Comment by bunnyyyyyy** [comment]
that is the code for the Graph.tsx. thanks

**Comment by vatsalajha** [comment]
You didn't make any changes in the code???

**Comment by bunnyyyyyy** [comment]
I didn't because they did not asked to change in before finishing up this step

**Comment by vatsalajha** [comment]
You need to, then run it

**Comment by bunnyyyyyy** [comment]
so what changes should I make there because it was not mentioned in the notes?

**Comment by vatsalajha** [comment]
It's mentioned in the slides, it's under the part name Making Changes in Main task panel. They have guided step by step , of what you need to do.

**Comment by bunnyyyyyy** [comment]
i only supposed to make the code changes once I am able to open the webpage according to the module. since my code is directing to the web[age but the page is empty, I am not able to proceed with it

**Comment by SiddhlodaAyyar** [comment]
Watch this video and if you still get the error comment down in the youtube section and we'll try to solve it.
https://youtu.be/wRAG9i8O_44

**Comment by bunnyyyyyy** [comment]
yes I have tried this way as well and it redirects to the website but the page is still blank. thanks
